src/pages/RepoPage/SettingsTab/tabs/BadgesAndGraphsTab/Badges/Badges.tsx

```diff
@@ -0,0 +1,150 @@
+import PropTypes from 'prop-types'
+import { useMemo, useState } from 'react'
+import { useParams } from 'react-router-dom'
+
+import config from 'config'
+
+import { Branch, useBranches } from 'services/branches'
+import A from 'ui/A'
+import Select from 'ui/Select'
+import SettingsDescriptor from 'ui/SettingsDescriptor'
+import TokenWrapper from 'ui/TokenWrapper'
+
+type UseBadgesProps = {
+  graphToken: string
+  provider: string
+  owner: string
+  repo: string
+  branch: Branch
+}
+
+const useBadges = ({
+  graphToken,
+  provider,
+  owner,
+  repo,
+  branch,
+}: UseBadgesProps) => {
+  const repoPath = `${config.BASE_URL}/${provider}/${owner}/${repo}`
+  let branchPath =
+    branch.name === 'Default branch'
+      ? ''
+      : `/branch/${encodeURIComponent(branch.name)}`
+  const fullPath = `${repoPath}${branchPath}/graph/badge.svg?token=${graphToken}`
+
+  const BadgesEnum = Object.freeze({
+    MARKDOWN: `[![codecov](${fullPath})](${repoPath})`,
+    HTML: `<a href="${repoPath}" > \n <img src="${fullPath}"/> \n </a>`,
+    RST: `.. image:: ${fullPath} \n :target: ${repoPath}`,
+  })
```

Comment:
`m`: Can we make this an `as const` object over `Object.freeze` (which actually only freezes the top level fields, and not the sub-fields)